src/app/contact-page/contact-page.component.ts

```diff
@@ -0,0 +1,79 @@
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import {Router} from '@angular/router';
+import {TicketService} from '../../services/ticket.service';
+import {AccountService} from '../../services/account.service';
+import {ModalContentComponent} from '../modal-content/modal-content.component';
+import {NgbModal, NgbModalRef} from '@ng-bootstrap/ng-bootstrap';
+
+
+@Component({
+  selector: 'app-contact-page',
+  templateUrl: './contact-page.component.html',
+  styleUrls: ['./contact-page.component.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class ContactPageComponent implements OnInit {
+  private submitDisabled: boolean;
+  private subject: string;
+  private message: string;
+
+  constructor(private router: Router, private accountService: AccountService, private ticketService: TicketService, private modalService: NgbModal) { }
+
+  private static validateEntry(entry: string, validator: (str: string) => boolean): boolean {
+    const trimmedEntry: string = (entry) ? entry.trim() : '';
+    return validator(trimmedEntry);
+  }
+
+
+  ngOnInit(): void {
+    this.accountService.authenticate(isAuthenticated => {
+      if (isAuthenticated) {
+        this.router.navigate(['support']);
+      } else {
+        this.router.navigate(['']);
+      }
+    });
+  }
+
+  private onSubmit(): void {
```

Comment:
This method is not needed if it does nothing else but call `this.createTicket()`.